src/actions.js

```diff
@@ -87,8 +87,44 @@ export function fetchOrganisations() {
       .then(responseObj => responseObj.json())
       .then(responseData => {
         const data = responseData.results;
+        const allOrganisations = data.map(organisation => {
+          //use organisation uuid without dashes only
+          return {
+            ...organisation,
+            uuid: organisation.uuid.replace(/-/g, "")
+          };
+        })
+        const availableOrganisations = allOrganisations.filter(e => {
+          return (
+            e.roles.find(e => e === "manager") ||
+            e.roles.find(e => e === "admin") ||
+            e.roles.find(e => e === "supplier") 
+            // users with only role "user" are no longer allowed in the management page. 
+            // ||
+            // e.roles.find(e => e === "user")
+          );
+        })
+
+        dispatch({ type: RECEIVE_ORGANISATIONS, all:allOrganisations, available: availableOrganisations});
 
-        dispatch({ type: RECEIVE_ORGANISATIONS, data });
+        if (
+          !organisation ||
```

Comment:
This variable was set in the top of this action creator (the first two lines), but here in the promise resolution the chosen organisation might have changed by now. I think it should be got from the state here.

Reply:
done moved to line 89